mongo/change_stream.go

```diff
@@ -457,20 +457,47 @@ func (cs *ChangeStream) ResumeToken() bson.Raw {
 }
 
 // Next gets the next result from this change stream. Returns true if there were no errors and the next
-// result is available for decoding.
+// result is available for decoding. Next blocks until an event is available for decoding or ctx expires.
+// If the given context expires during execution, the stream's error will be set and the change stream may be in an
+// invalid state and should be re-created. If Next returns false and an error occurred (i.e. cs.Err() != nil),
+// it must not be called again.
 func (cs *ChangeStream) Next(ctx context.Context) bool {
+	return cs.next(ctx, false)
+}
+
+// TryNext attempts to get the next result from this change stream. It returns true if there were no errors and the next
+// result is available for decoding. It returns false if the change stream was closed by the server, there was an
+// error getting more results from the server, or the server returned an empty batch of events. If an error occurred or
```

Comment:
Should applications check for error with `cs.Err()` instead?

Reply:
They should check for error with `cs.Err()`. What is the "instead" referring to?

Reply:
Instead of `cs.ID() == 0`

Reply:
that's referring to the part about the stream being closed/exhausted. i'll update it to check both